### PR TITLE
Refactor storage tests

### PR DESCRIPTION
**Pull request by wojtek-t**
Ref https://github.com/kubernetes/kubernetes/issues/109831

This further refactors the package to build a set of generic storage tests.

```release-note
NONE
```

/sig api-machinery
/kind cleanup
/priority important-longterm

/assign @stevekuznetsov @logicalhan @jpbetz

### CODE REVIEW

File: staging/src/k8s.io/apiserver/pkg/storage/etcd3/store_test.go [hunk 1]
```diff
@@ -152,31 +119,6 @@ func TestConditionalDelete(t *testing.T) {
 	storagetesting.RunTestConditionalDelete(ctx, t, store)
 }
 
-// The following set of Delete tests are testing the logic of adding `suggestion`
```
Comment (stevekuznetsov) [inline staging/src/k8s.io/apiserver/pkg/storage/etcd3/store_test.go:155]: I think we might want to keep this in the storagetesting package just for historical explanation?

**Review by stevekuznetsov — COMMENTED** [review]
One nit, otherwise LGTM - cancel hold as you'd like'

/lgtm
/hold

File: staging/src/k8s.io/apiserver/pkg/storage/etcd3/store_test.go (hunk 1 above)
Comment (wojtek-t) [reply staging/src/k8s.io/apiserver/pkg/storage/etcd3/store_test.go:155]: It's already there - we had two copies:
https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apiserver/pkg/storage/testing/store_tests.go#L273

I'm removing only the one from here, keeping the other one where the tests are actually implemented now.

Comment (stevekuznetsov) [reply staging/src/k8s.io/apiserver/pkg/storage/etcd3/store_test.go:155]: Sounds good!

**Review by logicalhan — COMMENTED** [review]
/lgtm

**Pull request merged**